tests/test_hf_mpt_gen.py

```diff
@@ -72,3 +78,108 @@ def test_init_hfhub_mpt(device: str, attn_impl: str):
 
 def test_init_hfhub_mpt_cpu():
     test_init_hfhub_mpt(device='cpu', attn_impl='torch')
+
+
+@pytest.mark.gpu
+@patch.object(ComposerGenerate,
+              'generate',
+              side_effect=ComposerGenerate.generate,
```

Comment:
oo this is my first time seeing side_effect...jw, what's the difference between using side_effect=ComposerGenerate.generate and wraps=ComposerGenerate.generate?

Reply:
also TIL about autospec=True..wish i knew about it sooner loll

Reply:
autospec is awesome 🙂 

Looks like theres a lot of questions about this (see comment) https://stackoverflow.com/questions/73086280/is-there-any-difference-between-using-mock-patch-object-with-wraps-vs-side-effec
🤷‍♀️ I think I can copied this from another example we had